### DISCUSSION TITLE
How to configure incremental selection with tree-sitter?

### DISCUSSION THREAD

**Discussion by mamachanko** [LunarVim/LunarVim#3835] [Q&A] [answered]
Editors like IntelliJ and vscode allow to incrementally expand selection. nvim-treesitter offers [incremental selection](https://github.com/nvim-treesitter/nvim-treesitter#incremental-selection).

However, I can't get it to work in a meaningful way.

My first attempt was:

```lua
lvim.builtin.nvimtree.setup.incremental_selection = {
    enable = true,
}
```

But the defaults to init and increment the selection are not working.

Then I tried different keymappings, but those did not work either:

```lua
lvim.builtin.nvimtree.setup.incremental_selection = {
    enable = true,
    keymaps = {
        init_selection = "<Leader>I",
        node_incremental = "<Leader>i",
        scope_incremental = "<Leader>o",
        node_decremental = "<Leader>O",
    },
}
```

Eventually, I tried
```lua
lvim.builtin.which_key.mappings["I"] = { "<cmd>lua require'nvim-treesitter.incremental_selection'.init_selection()<CR>", "incremental_selection" }
lvim.builtin.which_key.mappings["i"] = { "<cmd>lua require'nvim-treesitter.incremental_selection'.node_incremental()<CR>", "incremental_selection" }
```
That starts to select the current node on `<Leader>I` and increments once on `<Leader>i` but no further.

How have you configured incremental selection for an optimal experience?

**Comment by LostInTheLogs** [comment, marked as answer]
use `lvim.builtin.treesitter` not `lvim.builtin.nvimtree`

**Reply by mamachanko** [reply]
[lvim.builtin.treesitter](https://github.com/LunarVim/LunarVim/blob/1.2.0/lua/lvim/core/treesitter.lua#L5-L109) does neither expose `setup` nor `incremental_selection`. How would you configure it?

**Reply by LostInTheLogs** [reply]
use `lvim.builtin.treesitter.incremental_selection = {}`, incremental_selection isn't set by default so it doesn't appear in the autocompetion

**Reply by mamachanko** [reply]
That's it!

```lua
lvim.builtin.treesitter.incremental_selection = {
  enable = true,
}
```
^ enables incremental selection.

Thank you @LostNeophyte!
